src/containers/BooksList.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import Book from '../components/Book';
+
+function createTableHeader() {
+  return (
+    <tr>
+      <th>Book ID</th>
+      <th>Title</th>
+      <th>Category</th>
+    </tr>
+  );
+}
+
+function BooksList() {
+  const { books } = useSelector(state => state);
+  return (
+    <table>
+      <thead>
+        { createTableHeader() }
+      </thead>
+      <tbody>
+        { books.map(book => (
+          <Book
+            key={book.bookId}
+            id={book.bookId}
+            title={book.title}
+            category={book.category}
+          />
+        )) }
+      </tbody>
+    </table>
+  );
+}
+
+export default BooksList;
```

Comment:
You forgot the following
- connect to redux store ([see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/ConnetToRedux.png))

- use `mapStateToProps` to fetch books from the `store`

Reply:
I have implemented those, in the form of hooks:
* `useSelector`, please read more about it [here](https://react-redux.js.org/next/api/hooks). Here is a quick explanation in the documentation:
> The selector is approximately equivalent to the mapStateToProps argument to connect conceptually. The selector will be called with the entire Redux store state as its only argument. The selector will be run whenever the function component renders. useSelector() will also subscribe to the Redux store, and run your selector whenever an action is dispatched.

This requirement has been met in the form of hooks, please re-consider.

Reply:
thanks for enlightenment...i will definitely learn hooks today from your code  😊

Reply:
just correct, what's to be corrected and re-file for review. We'll re-consult the project and taking hooks into consideration 😊